api/ring-device.ts

```diff
@@ -32,15 +32,13 @@ export class RingDevice {
     return this.data.name
   }
 
-  get onComponentDevices() {
-    return this.location.onDevices.pipe(
-      reduce((acc, value) => {
-        const componentDevices = value.find(
-          (device) => device.data.parentZid === this.id
-        )
-        return acc.concat(componentDevices || [])
-      }, [] as RingDevice[])
-    )
+  getComponentDevices(callback: (componentDevices: RingDevice[]) => void) {
+    return this.location.onDevices.subscribe((devices) => {
+      const componentDevices = devices.filter(
+        ({ data }) => data.parentZid === this.id
+      )
+      callback(componentDevices)
+    })
```

Comment:
If you want to go with `getComponentDevices` instead of the `onComponentDevices` observable (which is totally fine), I'd prefer that it use an async function instead of a callback.  

```suggestion
  async getComponentDevices(callback: (componentDevices: RingDevice[]) => void) {
    const devices = await this.location.getDevices()
    return componentDevices = devices.filter(
        ({ data }) => data.parentZid === this.id
      )
```

Reply:
Makes sense, but I switched back to an `Observable`-returning `onComponentDevices`. The `Observable`-based API seemed more consistent with the classes, and could be invoked easily in the `Thermostat` constructor (no need for a `;(async () => {})()`-style [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE)).